src/PMGGenverter.cpp

```diff
@@ -0,0 +1,99 @@
+#include <power_whisperpower/PMGGenverter.hpp>
+
+using namespace power_whisperpower;
+using protocol::fromBigEndian;
+
+PMGGenverter::PMGGenverter()
+{
+}
+
+void PMGGenverter::process(canbus::Message const& msg)
+{
+    switch (msg.can_id) {
+        case 0x200:
+            m_status.time = base::Time::now();
+            m_status.ac_voltage =
+                static_cast<float>(fromBigEndian<uint16_t>(msg.data)) / 10;
+            m_status.ac_current =
+                static_cast<float>(fromBigEndian<uint16_t>(msg.data + 2)) / 10;
+            m_status.engine_angular_speed =
+                static_cast<float>(fromBigEndian<uint16_t>(msg.data + 4)) * 2 * M_PI / 60;
+            m_status.inverter_temperature = base::Temperature::fromCelsius(
+                fromBigEndian<uint16_t>(msg.data + 6) / 10.0);
+            break;
+        case 0x201:
+            m_status.status = fromBigEndian<uint16_t>(msg.data);
+            m_status.inverter_alarm = fromBigEndian<uint16_t>(msg.data + 2);
+            m_status.inverter_warning = fromBigEndian<uint16_t>(msg.data + 4);
+            m_status.engine_alarm = fromBigEndian<uint16_t>(msg.data + 6);
+            break;
+        case 0x202:
+            m_status.stepper = static_cast<float>(fromBigEndian<uint16_t>(msg.data));
+            m_status.oil_temperature = base::Temperature::fromCelsius(
+                fromBigEndian<uint16_t>(msg.data + 2) / 10.0);
+            m_status.delta_dc_bus =
+                static_cast<float>(fromBigEndian<uint16_t>(msg.data + 4));
+            m_status.PWM_scale =
+                static_cast<float>(fromBigEndian<uint16_t>(msg.data + 6));
+            break;
+        case 0x203:
+            m_status.TEST_ramp = msg.data[7];
+            break;
+        case 0x204:
+            m_status.inverter_model =
+                std::string(1, msg.data[0]) + std::string(1, msg.data[1]);
+            m_status.firmware_version = static_cast<int>(msg.data[2]);
+            m_status.firmware_subversion = static_cast<int>(msg.data[3]);
+            m_status.hardware_version = static_cast<int>(msg.data[4]);
+            m_status.hardware_subversion = static_cast<int>(msg.data[5]);
```

Comment:
Converting those to int *here* is a waste of space. Keep uint8_t in the data structure. You need to convert to int when displaying, or iostream interprets it as a character (and shows garbage)